docs/advanced.rst

```diff
@@ -0,0 +1,208 @@
+    class CachedData:
+
+        key: str
+        value: str
+
+
+    @provides(CachedData)
+    class CachedDataProvider(PageObjectInputProvider):
+
+        def __init__(self, response: DummyResponse):
+            self.response = response
+
+        def __call__(self):
+            return CachedData(
+                key=self.response.url,
+                value=get_cached_content(self.response.url)
+            )
+
+
+    class MyPageObject(ItemPage):
+
+        content: CachedData
+
+        def to_item(self):
+            return {
+                'url': self.content.key,
+                'content': self.content.value,
+            }
+
+
+    class MySpider(scrapy.Spider):
+
+        name = 'my_spider'
+
+        def parse(self, response: DummyResponse, page: MyPageObject):
+            # request will be IGNORED because neither spider callback
+            # not MyPageObject seem like to be making use of its response
+            yield page.to_item()
+
+Although, if the spider callback is not going to use the ``Response``, but the
+Page Object makes use of it, the request is not going to be ignored and will be
+processed, for example:
+
+.. code-block:: python
+
+    def parse_content(html: str):
+        # parse content from html
+        pass
+
+
+    @attr.s(auto_attribs=True)
+    class MyResponseData:
+
+        url: str
+        html: str
+
+
+    @provides(MyResponseData)
+    class MyResponseDataProvider(PageObjectInputProvider):
+
+        def __init__(self, response: Response):
+            self.response = response
+
+        def __call__(self):
+            return MyResponseData(
+                url=self.response.url,
+                html=self.response.content,
+            )
+
+
+    class MyPageObject(ItemPage):
+
+        response: MyResponseData
+
+        def to_item(self):
+            return {
+                'url': self.response.url,
+                'content': parse_content(self.response.html),
+            }
+
+
+    class MySpider(scrapy.Spider):
+
+        name = 'my_spider'
+
+        def parse(self, response: DummyResponse, page: MyPageObject):
+            # request will be PROCESSED because spider callback is not
+            # making use of its response, but MyPageObject seems like to be
+            yield page.to_item()
+
+.. note::
+
+    The code above is just for example purposes. If you need to use ``Response``
+    instances in your code, make use of ``ItemWebPage`` as it makes use of the
+    built-ins ``ResponseData`` and ``ResponseDataProvider``.
+
+Requests concurrency
+--------------------
+
+DummyRequests are meant to skip downloads, so it makes sense not checking for
+concurrent requests, delays, or auto throttle settings since we won't be making
+any download at all.
+
+By default, if your parser and its page inputs need a regular Request, it will
+be downloaded and all of those Scrapy settings related to it are going to be
+respected, for example:
+
+- ``CONCURRENT_REQUESTS``
```

Comment:
If I'm not mistaken, this option is still respected with scrapy-poet (/cc @ivanprado).

Reply:
Yes, it is still respected even when dummy request is used.

Reply:
Yes, it's respected. Docs says:

- by default, those settings are respected
- be aware when using third-party libraries for requests because they're probably ignoring these settings

Just double-checking, is any action needed here, @kmike? Any suggestions on how to clarify this passage?

Reply:
If you make an HTTP request in a provider using some third-party async library (aiohttp, treq, etc.), CONCURRENT_REQUESTS option will be respected, but not the other settings. 

To have other settings respected, in addition to CONCURRENT_REQUESTS, you'd need to use crawler.engine.download or something like that; alternatively, implement limits in a library.

We'll also make it possible to use DownloaderAwarePriorityQueue in such cases, by refactoring Scrapy (this is a separate task).

Reply:
I've just updated the docs. Would you mind having another look? If you're cool with the latest modifications, we could go ahead and merge this PR.